### PR TITLE
Reorganize daqconf schema options

### LINKED ISSUE #345: Reorganize daqconf schema options
The daqconf schema (https://github.com/DUNE-DAQ/daqconf/blob/develop/schema/daqconf/confgen.jsonnet) was originally split into groups in a fairly arbitrary way. The parameters should be reorganized to ensure that app-specific parameters are grouped together, and possibly add additional groups where needed for cross-app parameters (instead of putting most of them in `boot`).

### PR DESCRIPTION
**Pull request by alessandrothea**
This PR implements an initial re-organisation of the confgen parameters described in #345 

- Adds a detector `detector` record for detector/apparatus/etc global parameters
- Adds a `daq_common` record for global daq parameters shared across applications
- Updates some of the parameters for better readability (e.g. `boot.use_k8s` -> `boot.process_manager`)

It also reorganise (parts) of the daqconf_multi_ru code into subfunctions.
This PR supersedes #330

Furthermore, this PR adds the following commandline options to `daqconf_multiru_gen`:
```
--force-pm [ssh|k8s]                    Force process manager
 -n, --dry-run                                 Dry run, do not generate output files
 -a, --check-args-and-exit           Check input arguments and quit
  -f, --force                                     Force configuration generation - delete existing target directory if exists
```

Also added schema checker utility script `daqconf_check_schema/py`

### CODE REVIEW

File: schema/daqconf/confgen.jsonnet [hunk 1]
```diff
@@ -31,42 +32,49 @@ local cs = {
...
+    s.field( "data_request_timeout_ms", self.count, default=1000, doc="The baseline data request timeout that will be used by modules in the Readout and Trigger subsystems (i.e. any module that produces data fragments). Downstream timeouts, such as the trigger-record-building timeout, are derived from this."),
     s.field( "use_data_network", self.flag, default = false, doc="Whether to use the data network (Won't work with k8s)"),
-  ]),
+    s.field( "data_rate_slowdown_factor",self.count, default=1, doc="Factor by which to suppress data generation. Former -s"),
+  ], doc="Cmmon daq_common settings"),
+
+  detector :  s.record("detector", [
+    s.field( "op_env", self.string, default='swtest', doc="Operational environment - used for HDF5 Attribute inside the files"),
+    s.field( "clock_speed_hz", self.freq, default=62500000),
+    s.field( "tpg_channel_map", self.tpg_channel_map, default="PD2HDChannelMap", doc="Channel map for TPG"),
```
Comment (plasorak) [inline schema/daqconf/confgen.jsonnet]: You have changed the default channel map, right?

**Comment by plasorak** [comment]
I've just tested that with a slightly modified version of integrationtest: https://github.com/DUNE-DAQ/integrationtest/pull/42 and dfmodules: https://github.com/DUNE-DAQ/dfmodules/pull/297 and it works. I'd be inclined to merge it.

File: schema/daqconf/confgen.jsonnet (hunk 1 above)
Comment (alessandrothea) [reply schema/daqconf/confgen.jsonnet]: I have indeed. It used to be

```s.field( "tpg_channel_map", self.tpg_channel_map, default="ProtoDUNESP1ChannelMap", doc="Channel map for TPG"),```

on the trigger configuration record.

Considering that there are no more ProtoWIBs around, `PD2HDChannelMap`  seemed more appropriate...

Comment (alessandrothea) [reply schema/daqconf/confgen.jsonnet]: Hm, this made me realise I should have changed the name to `tpc_channel_map`, damn.

**Review by bieryAtFnal — APPROVED** [review]
Tests so far look good to me.

**Pull request merged**